sync/src/types.rs

```diff
@@ -211,25 +211,45 @@ impl PeerState {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Default)]
+pub struct Filter<T: Eq + Hash> {
+    inner: LruCache<T, ()>,
+}
+
+impl<T: Eq + Hash> Filter<T> {
+    pub fn new(size: usize) -> Self {
+        Self {
+            inner: LruCache::new(size),
+        }
+    }
+
+    pub fn contains(&self, item: &T) -> bool {
+        self.inner.contains_key(item)
```

Comment:
https://github.com/contain-rs/lru-cache/blob/master/src/lib.rs#L100
https://github.com/contain-rs/lru-cache/blob/master/src/lib.rs#L148

I am not sure whether it should refresh order.

https://github.com/nervosnetwork/lru-cache/blob/master/src/lib.rs#L42